api/http/indigo_service/indigo_http.py

```diff
@@ -0,0 +1,216 @@
+#
+# Copyright (C) from 2009 to Present EPAM Systems.
+#
+# This file is part of Indigo toolkit.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Awaitable, Callable, List, Tuple, Union
+
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse
+from indigo import IndigoException
+
+from indigo_service import jsonapi, service
+from indigo_service.indigo_tools import indigo, indigo_new
+
+app = FastAPI(title="Indigo service JSON:API", version=indigo().version())
+
+
+BASE_URL_INDIGO = "/indigo"
+RESP_HEADER_CONTENT_TYPE = "application/vnd.api+json"
+
+
+@app.middleware("http")
+async def isolate_indigo_session(
+    request: Request, call_next: Callable[[Request], Awaitable[JSONResponse]]
+) -> JSONResponse:
+    with indigo_new():
+        response = await call_next(request)
+        if not request.scope["path"].startswith(
+            ("/docs", "/redoc", f"{BASE_URL_INDIGO}/render")
+        ):
+            response.headers["Content-Type"] = RESP_HEADER_CONTENT_TYPE
+        return response
+
+
+@app.exception_handler(IndigoException)
+def unicorn_exception_handler(
+    request: Request, exc: IndigoException  # pylint: disable=unused-argument
+) -> JSONResponse:
+    error = jsonapi.make_error_response(exc)
+    return JSONResponse(status_code=400, content=error.dict())
```

Comment:
Should we also add a general exception handler for `Exception` and return 500 in that case?

Reply:
500 by Exception is a default behaviour